src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,43 @@
+import { getArticleById } from "../utils/api";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [comments, setComments] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  let formattedDate = new Date(article.created_at).toLocaleDateString("en-us", {
```

Comment:
Given that you are using the same date formatting behaviour here - more reason to extract to reuse elsewhere